hazelcast/src/main/java/com/hazelcast/client/impl/proxy/ClientMapProxy.java

```diff
@@ -1984,4 +1988,39 @@ protected void onDestroy() {
             super.onDestroy();
         }
     }
+
+    @Override
+    public V computeIfPresent(K key, BiFunction<? super K, ? super V, ? extends V> remappingFunction) {
+        checkNotNull(key, NULL_KEY_IS_NOT_ALLOWED);
+        checkNotNull(key, NULL_BIFUNCTION_IS_NOT_ALLOWED);
+
+        if (SerializationUtil.isClassStaticAndSerializable(remappingFunction)) {
```

Comment:
This is where it gets tricky again. The client is not aware of the version of the cluster it is connected to. So it might happen that the function is serializable and static and is sent to the cluster but the cluster members fail to deserialise it. Unfortunately, this would mean there is no other option for the user but to not have a serializable function when calling from the client or to upgrade the member versions.

I don't have a good idea how to tackle this. As a poor-man's solution (but, still better than an infinite loop which we have now), we can avoid this optimisation on the client and simply use `computeIfPresentLocally` in all cases. WDYT? cc @sancar

Reply:
Hi, @mmedenjak  is right we need a fallback mechanism when server is not available. Having a fallback mechanism have greater scope. Lets make the `computeIfPresentLocally` only implementation for now for the client. The optimisation for the client can be addressed later.

Reply:
Thanks @mmedenjak @sancar! I will push a commit to address all feedback.